### DISCUSSION TITLE
aiogram and aiomysql

### DISCUSSION THREAD

**Discussion by AnatoliiA** [aiogram/aiogram#568] [General]
## Context

Please provide any relevant information about your setup. This is important in case the issue is not reproducible except for under certain conditions.

* Operating System: ubuntu
* Python Version:  3.7 
* aiogram version: 2.11.2
* aiohttp version: no 
* uvloop version (if installed):  no

## Expected Behavior

I expected getting a data from the mysql base used aiomysql
im async way for aiogram bot:

```python
import asyncio, aiomysql
async def execute(loop, sttr) :
    connection(*param)
    cur = await conn.cursor()
    await cur.execute(sttr)
    result = await cur.fetchone()
    conn.close()

#  handler catch update
@dp.message_handler(state=docregistration.name)
async def enter_name(message: types.Message, state: FSMContext):
    task = loop.create_task(execute_one(sttr))
    p = loop.run_until_complete(task)
    loop.close()
```
## Current Behavior
raise RuntimeError('This event loop is already running')
RuntimeError: This event loop is already running
<b>Because function enter_name() finished before complete event loop even though sleep()<b>

<H1><i>Any suggestion when I wrong</i></H1>

**Comment by Olegt0rr** [comment]
It's not relevant to aiogram at all.

Please, use asyncio docs to learn how to use task properly, or ask stackoverflow for that case.

Hints:
You shouldn't use loop directly.
Use await to await task.
Don't create/close connection on every update.

**Reply by AnatoliiA** [reply]
You shouldn't use loop directly.  
I can not understand.

Could you help me?

**Reply by Olegt0rr** [reply]
In this case you shouldn't use the `loop` at all.
It's a async programming basics, please read the python docs.
I can't teach you right here. Use google and stackoverflow to find an answers.

**Reply by AnatoliiA** [reply]
Thank you.
